src/layout/rectangles.js

```diff
@@ -0,0 +1,121 @@
+import d3 from 'd3';
+import dataJoinUtil from '../util/dataJoin';
+import {noop} from '../util/fn';
+import {range} from '../util/scale';
+import noopStrategy from './strategy/noop';
+import {rebindAll} from '../util/rebind';
+
+export default function(layoutStrategy) {
+
+    var width = d3.functor(0),
+        height = d3.functor(0),
+        x = function(d, i) { return d.x; },
+        y = function(d, i) { return d.y; };
+
+    var xScale = d3.scale.identity(),
+        yScale = d3.scale.identity(),
+        strategy = layoutStrategy || noopStrategy(),
+        component = noop;
+
+    var dataJoin = dataJoinUtil()
+        .selector('g.rectangle')
+        .element('g')
+        .attr('class', 'rectangle');
+
+    var rectangles = function(selection) {
+
+        var xRange = range(xScale),
+            yRange = range(yScale);
+        strategy.containerWidth(Math.max(xRange[0], xRange[1]))
+            .containerHeight(Math.max(yRange[0], yRange[1]));
+
+        selection.each(function(data, index) {
+            var g = dataJoin(this, data);
```

Comment:
Hm this is another implementation of dataJoin on one dataset but use properties from another. It's not exactly the same as multi but I do wonder whether it's worth considering breaking that out from multi in the future.